src/jsm.c

```diff
@@ -3083,7 +3105,7 @@ js_AddConsumer(jsConsumerInfo **new_ci, jsCtx *js,
         {
             // No subject filter, use <stream>.<consumer name>
             // otherwise, the filter subject goes at the end.
-            if (nats_IsStringEmpty(cfg->FilterSubject))
+            if (nats_IsStringEmpty(cfg->FilterSubject) || (cfg->FilterSubjectsLen > 0))
```

Comment:
Probably an unnecessary check. When both `filter_subject` and `filter_subjects` are set server seems to be returning `consumer cannot have both` error whether the API subject was appended with `filter_subject` or not.

Reply:
took it out